src/twisted/plugins/twisted_reactors.py

```diff
@@ -4,59 +4,70 @@
 from __future__ import absolute_import, division
 
 from twisted.application.reactors import Reactor
-from twisted.python.compat import _PY3
+from twisted.python.reflect import requireModule
 
-default = Reactor(
+def _addReactorToAll(shortName, moduleName, description):
+    """
+
+    @param shortName: Short name of reactor
+    @param moduleName: The fully-qualified module name of the reactor
+    @param description: Description of reactor
+    @return: newly created reactor
+    @rtype: L{Reactor} or L{None}
+    """
+    newReactor = None
+    if requireModule(moduleName) is not None:
```

Comment:
Maybe.  The trouble is just that the intent of the PR was far from clear and this confused a lot of the discussion that resulted.

So, back to technical matters.  The point about dropin.cache not being regenerated means that this change will make `--help-reactors` wrong in a different way (I commented on the Twisted ticket about this a few days ago).

A different approach that won't suffer from this problem is to define all of the reactors in the dropin and then filter them down to those that actually work at some later point (probably in some part of the `--help-reactors` implementation).